Automattic/jetpack
# Donations block: Filter the donation subscriptions based on public property

Fixes [#61908](https://github.com/Automattic/wp-calypso/issues/61908)

#### Changes proposed in this Pull Request:

* Filter the donation subscription by public false to keep the existing behaviour of donation subscriptions

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Apply the diffs on your sandbox: `arc patch D77469 && arc patch D77465 --nobranch && arc patch D77864 --nobranch`
* Create a new site (so that you don't have any existing donation subscription that are hidden)
* Atomic: You'll need to use your WoA dev site and merge locally the branch from https://github.com/Automattic/jetpack/pull/23714 and sync your WoA site with these changes
* Atomic: add `define( 'JETPACK__SANDBOX_DOMAIN', 'your-sandbox');` to your wp-config.php
* Use the Calypso link to create a new donation subscription in `/earn/payments-plans/` from this [PR](https://github.com/Automattic/wp-calypso/pull/62208)
* Go to wp-admin/post-new.php and create a new donation block
* Using Network inspector, check if the donation block DOESN'T use the subscription you've created in `/earn`
